src/protocol.rs

```diff
@@ -386,21 +386,29 @@ pub(crate) mod participant_config {
 
 /// An identifier for a [`Participant`].
 ///
-/// All [`Participant`]s in a session must agree on the
+/// All [`Participant`]s in a sub-protocol must agree on the
 /// [`ParticipantIdentifier`]s. That is, these are not local identifiers
 /// controlled by a single `Participant`; they are unique, agreed-upon
-/// identifiers for the `Participant`s in a session. Each entity participating
-/// in a session should have a different `ParticipantIdentifier`.
+/// identifiers for the `Participant`s executing a sub-protocol. Each member
+/// participating in a sub-protocol should have a different `ParticipantIdentifier`.
 ///
-/// `ParticipantIdentifier`s can be used across multiple sessions. For
-/// example, if a set of participants run keygen, auxinfo, and then compute
-/// several signatures, they can use the same set of identifiers for each of
-/// those sessions. However, a single `ParticipantIdentifier` should not be used
-/// to represent different entities (even in different sessions with
+/// The same `ParticipantIdentifier`s are used across multiple sub-protocols. For
+/// example, if a set of participants run keygen, aux-info, and then compute
+/// several signatures, they must use the same set of identifiers for each of
+/// those sessions. These `ParticipantIdentifier`s are embedded in the output
+/// of certain sub-protocols, e.g. aux-info: this library will check the expected
+/// `ParticipantIdentifier` from the aux-info output against the `ParticipantIdentifier`
+/// of the `Participant` object executing the sub-protocol. An error will be returned
+/// if they don't match. Therefore, it is up to the calling application to remember and
+/// use the expected `ParticipantIdentifier` when executing sub-protocols.
+///
+/// However, a single `ParticipantIdentifier` should not be used
```

Comment:
However is not needed on this line, but otherwise looks good

Reply:
Fixed!